linker/shared/src/main/scala/org/scalajs/linker/analyzer/Infos.scala

```diff
@@ -544,9 +544,18 @@ object Infos {
 
             case IsInstanceOf(_, testType) =>
               builder.maybeAddUsedInstanceTest(testType)
+
             case AsInstanceOf(_, tpe) =>
               builder.maybeAddUsedInstanceTest(tpe)
 
+              /* In theory, we'd need to reach ClassCastException
```

Comment:
This reads as if it applies to nodes *below* the comment, despite the indentation.
Consider moving that right after the line `case AsInstanceOf(_, tpe) =>`.